ls1intum/Artemis
# `Adaptive learning`: Add legend to learning path graph representation

### Checklist
#### General


#### Client

### Motivation and Context



To help users to better understand the learning path graph representation there should be a legend to better describe the meaning of the used icons.

Resolves #7175 

### Description

This PR adds a new component containing the legend.

### Steps for Testing

Prerequisites:
- 1 Students/Instructor
- 1 Course with learning paths enabled

1. Log in to Artemis
2. Navigate to a student's learning path and open the graph view
3. Make sure there is a legend in the bottom right corner of the graph
4. Make sure the translations are correctly displayed
5. Hover over the legend and check if the tooltips are displayed and correctly translated
6. Make sure the legend is readable in dark and light mode


### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage







#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [learning-path-legend.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5505/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/learning-paths/learning-path-graph/learning-path-legend.component.ts.html) | 100% | ✅ |

�

### Screenshots


<img width="1131" alt="Screenshot 2023-09-28 at 15 16 10" src="https://github.com/ls1intum/Artemis/assets/44003963/07bcf060-fce3-4984-b5b2-7222e3390288">

<img width="307" alt="Screenshot 2023-09-28 at 15 16 31" src="https://github.com/ls1intum/Artemis/assets/44003963/c978bc0a-9dc6-47ae-a8d4-67958fd99ee0">

<img width="279" alt="Screenshot 2023-09-28 at 15 16 47" src="https://github.com/ls1intum/Artemis/assets/44003963/137fff16-13f1-480b-b025-edc98ad427a6">

<img width="308" alt="Screenshot 2023-09-28 at 15 17 01" src="https://github.com/ls1intum/Artemis/assets/44003963/05c08fb2-51cd-4316-a046-f78231cbb8af">
